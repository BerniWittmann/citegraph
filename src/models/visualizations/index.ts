import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
