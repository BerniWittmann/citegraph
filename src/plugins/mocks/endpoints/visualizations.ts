import { AxiosRequestConfig } from 'axios'
import MockAdapter from 'axios-mock-adapter'
import db from '../db'
import Project from '@/models/project'

const projects = db.getCollection('projects')
const visualizations = db.getCollection('visualizations')

function getProjectId (config: AxiosRequestConfig): number {
  const urlParts: Array<string> = config.url!.split('/')
  return parseInt(urlParts[4])
}

export function setupMocks (mock: MockAdapter) {
  mock.onGet(/\/projects\/\d+\/visualizations$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    return [200, visualizations.find()]
  })
}
