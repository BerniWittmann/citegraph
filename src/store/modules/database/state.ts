import { DatabaseState } from './types'

export const state: DatabaseState = {}
