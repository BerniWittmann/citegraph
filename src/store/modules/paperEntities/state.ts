import { PaperEntitiesState } from './types'

export const state: PaperEntitiesState = {
  entityType: undefined,
  entities: [],
  entityCount: -1
}
