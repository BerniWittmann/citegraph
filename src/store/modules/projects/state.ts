import { ProjectsState } from './types'

export const state: ProjectsState = {
  activeProject: undefined,
  projects: [],
  openProjects: []
}
