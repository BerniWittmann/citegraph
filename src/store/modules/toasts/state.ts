import { ToastsState } from './types'

export const state: ToastsState = {
  toasts: []
}
