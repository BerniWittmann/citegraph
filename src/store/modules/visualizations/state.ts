import { VisualizationsState } from './types'

export const state: VisualizationsState = {
  visualizations: []
}
