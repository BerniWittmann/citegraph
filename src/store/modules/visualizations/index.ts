import { Module } from 'vuex'
import { VisualizationsState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const visualizations: Module<VisualizationsState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
