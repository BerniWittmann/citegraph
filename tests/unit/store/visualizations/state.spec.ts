import { state } from '@/store/modules/visualizations/state'

describe('store/modules/visualizations/state', () => {
  it('provides the initial state', () => {
    expect(state).toMatchSnapshot()
  })
})
