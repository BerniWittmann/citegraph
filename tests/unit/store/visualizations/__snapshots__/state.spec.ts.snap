// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/modules/visualizations/state provides the initial state 1`] = `
Object {
  "visualizations": Array [],
}
`;
